src/osal/osal.h

```diff
@@ -40,9 +40,10 @@ enum
 {
   OSAL_TIMEOUT_NOTIMEOUT    = 0,      // return immediately
   OSAL_TIMEOUT_NORMAL       = 10,     // default timeout
-  OSAL_TIMEOUT_WAIT_FOREVER = 0xFFFFFFFFUL
 };
 
+static const uint32_t OSAL_TIMEOUT_WAIT_FOREVER = 0xFFFFFFFFUL;

```

Comment:
the 0xFFFFFFFFUL was being interpreted as signed, which was above the max value of a signed long.... only way around it is to declare it as a const int. I don't like it, though.